### ISSUE TITLE
Validation isNumber() can accept empty space

### ISSUE THREAD

**Issue by Jessy-BAER** [ivan-dalmet/formiz#10]
It could be nice if `isNumber` can accept empty space, because for now we need to use custom validation to do that.

![image (1)](https://user-images.githubusercontent.com/7570753/72969522-ec44b980-3dc5-11ea-929d-57ae17de571d.png)

**Comment by yoannfleurydev** [comment]
I think we should introduce a new validation for this kind of wanted behavior: `isOptionalNumber` or something like that. So `isNumber` stays the same.

**Comment by ivan-dalmet** [comment]
That was my first idea but in reality, all rules should accept "empty"  (`null`, `undefined`, `""`) value.
Because this rule is handle with the `isRequired` rule (or `required` property).

Example for optional field:

Now:
```JSX
<Field
  name="myField"
  validations={[
    {
      rule: val => !value && value !== 0 ? true : isNumber()(value)
      message: 'not a number',
    },
    {
      rule: val => !value && value !== 0 ? true : isMinNumber(1)(value)
      message: 'not under 1',
    },
    {
      rule: val => !value && value !== 0 ? true : isMaxNumber(10)(value)
      message: 'not over 10',
    },
  ]}
>
```

If we allow empty value:
 validations={[
    {
      rule: isNumber()
      message: 'not a number',
    },
    {
      rule: isMinNumber(1)
      message: 'not under 1',
    },
    {
      rule: isMaxNumber(10)
      message: 'not over 10',
    },
  ]}
>
```

**Comment by ivan-dalmet** [comment]
Available in version `1.0.0-rc.4` 🙂
